spec/models/regulator_spec.rb

```diff
@@ -92,4 +106,53 @@
       end
     end
   end
+
+  describe "#batch_inspect!" do
+    context "with a few records" do
+      before do
+        ActiveRecord::Base.transaction do
+          10.times { create(:reading, :day_time, {temp: 75, outdoor_temp: 50 }) }
+        end
+      end
+
+      it "updates violation status in batches", :vcr do
+        expect(Reading.where(violation: true)).to have(0).items
+
+        Reading.order(created_at: :asc).first.update(temp: 65)
+        Reading.order(created_at: :asc).last.update(temp: 65)
+
+        Regulator.new(Reading.all).batch_inspect!(silent: true)
+        expect(Reading.where(violation: true)).to have(2).items
+      end
+    end
+
+    context "with many records" do
```

Comment:
I see, a stress test. Not sure it's useful to have it in a test suite, since the test env is very different from production. Running it as a rake task against a copy of production might be a closer bet... Still, it wouldn't be an active app with possible concurrent requests and db writes, etc. I find New Relic is great for this sort of thing, since it gives you an xray of real activity on your app. I didn't find any NR config on the app, so I'm assuming it's not using it yet.

By the way, have you looked at [Model.find_in_batches](http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_in_batches)? And [Model#update_columns](http://apidock.com/rails/v4.2.1/ActiveRecord/Persistence/update_columns) might be good for faster saves. Just remember to also update the `updated_at` column